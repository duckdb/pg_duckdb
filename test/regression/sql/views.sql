create table view_table(a varchar);
insert into view_table values ('test'), ('hello');

create view vw as select * from view_table;

select * from vw;
select * from vw offset 1;
select * from vw limit 1;

drop view vw;

create schema s;
create table s.t as select 21;
create table "s.t" as select 42;

create view vw1 as select * from s.t;
create view vw2 as select * from "s.t";

select * from vw1, vw2;

drop view vw1;
drop view vw2;
drop table "s.t";
drop table s.t;
drop schema s;
drop table view_table;
