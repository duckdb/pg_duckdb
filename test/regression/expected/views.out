create table view_table(a varchar);
insert into view_table values ('test'), ('hello');
create view vw as select * from view_table;
select * from vw;
   a   
-------
 test
 hello
(2 rows)

select * from vw offset 1;
   a   
-------
 hello
(1 row)

select * from vw limit 1;
  a   
------
 test
(1 row)

drop view vw;
create schema s;
create table s.t as select 21;
create table "s.t" as select 42;
create view vw1 as select * from s.t;
create view vw2 as select * from "s.t";
select * from vw1, vw2;
WARNING:  (PGDuckDB/CreatePlan) Prepared query returned an error: 'Binder Error: Referenced table "t" not found!
Candidate tables: "s.t"
LINE 1: ...?column?", vw2."?column?" FROM (SELECT t."?column?" FROM s.t) vw1, (SELECT "s....
                                                  ^
 ?column? | ?column? 
----------+----------
       21 |       42
(1 row)

drop view vw1;
drop view vw2;
drop table "s.t";
drop table s.t;
drop schema s;
drop table view_table;
